### PR TITLE
Fix hidden link section in Menu Item modal

### PR DESCRIPTION
**Pull request by kamilpastuszka**
I want to merge to fix visibility issue in Menu Item modal.

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://master.staging.saleor.cloud/graphql/

### CODE REVIEW

**Review by orzechdev — APPROVED** [review]

File: src/components/AutocompleteSelectMenu/AutocompleteSelectMenu.tsx [hunk 1]
```diff
@@ -80,7 +80,7 @@ const AutocompleteSelectMenu: React.FC<AutocompleteSelectMenuProps> = props => {
   } = props;
   const classes = useStyles(props);
 
-  const [inputValue, setInputValue] = React.useState(displayValue || "");
+  const [inputValue, setInputValue] = React.useState(displayValue || null);
```
Comment (jwm0) [inline src/components/AutocompleteSelectMenu/AutocompleteSelectMenu.tsx:83]: Is this change linked with the task in any way? If not, using `""` as default value for input does not produce any warnings in console

File: src/navigation/components/MenuItemDialog/MenuItemDialog.tsx [hunk 2]
```diff
@@ -238,7 +238,7 @@ const MenuItemDialog: React.FC<MenuItemDialogProps> = ({
               id: "menuItemDialogAddItem"
             })}
       </DialogTitle>
-      <DialogContent style={{ overflowY: "visible" }}>
+      <DialogContent style={{ overflowY: "visible", overflowX: "visible" }}>
```
Comment (jwm0) [inline src/navigation/components/MenuItemDialog/MenuItemDialog.tsx:241]:
```suggestion
      <DialogContent style={{ overflow: "visible" }}>
```

**Review by jwm0 — CHANGES_REQUESTED** [review]

File: src/components/AutocompleteSelectMenu/AutocompleteSelectMenu.tsx (hunk 1 above)
Comment (kamilpastuszka) [reply src/components/AutocompleteSelectMenu/AutocompleteSelectMenu.tsx:83]: Leaving  `""` there is the reason of the issue with placeholder text related to onBlur event. This is the second bug related to this PR.
I believe this part of the code is to handle no value case, not to deal with warnings or errors.

Comment (jwm0) [reply src/components/AutocompleteSelectMenu/AutocompleteSelectMenu.tsx:83]: This fixes one bug but creates another, try to pick an item without typing anything into the Input and you'll see that the issue still occurs.

**Review by jwm0 — CHANGES_REQUESTED** [review]

File: src/components/AutocompleteSelectMenu/AutocompleteSelectMenu.tsx [hunk 3]
```diff
@@ -123,7 +123,6 @@ const AutocompleteSelectMenu: React.FC<AutocompleteSelectMenuProps> = props => {
                   onBlur: () => {
                     closeMenu();
                     setMenuPath([]);
-                    setInputValue(displayValue);
```
Comment (jwm0) [inline src/components/AutocompleteSelectMenu/AutocompleteSelectMenu.tsx:126]: This serves a purpose here, we want to fallback to the selected value on blur if user clears the input.

**Review by jwm0 — CHANGES_REQUESTED** [review]

**Review by jwm0 — APPROVED** [review]

**Pull request merged**